src/Core/Services/GraphQLSchemaCreator.cs

```diff
@@ -251,6 +266,252 @@ private DocumentNode GenerateSqlGraphQLObjects(RuntimeEntities entities, Diction
+        /// <param name="referencingColumns">Referencing columns in the relationship.</param>
+        private static void AddReferencingFieldDirectiveToReferencingFields(
+            Dictionary<string, FieldDefinitionNode> referencingEntityFieldDefinitions,
+            List<string> referencingColumns,
+            ISqlMetadataProvider metadataProvider,
+            string entityName)
+        {
+            foreach (string referencingColumn in referencingColumns)
+            {
+                if (metadataProvider.TryGetExposedColumnName(entityName, referencingColumn, out string? exposedReferencingColumnName) &&
+                    referencingEntityFieldDefinitions.TryGetValue(exposedReferencingColumnName, out FieldDefinitionNode? referencingFieldDefinition))
+                {
+                    if (!referencingFieldDefinition.Directives.Any(directive => directive.Name.Value == ReferencingFieldDirectiveType.DirectiveName))
+                    {
+                        List<DirectiveNode> directiveNodes = referencingFieldDefinition.Directives.ToList();
+                        directiveNodes.Add(new DirectiveNode(ReferencingFieldDirectiveType.DirectiveName));
+                        referencingEntityFieldDefinitions[exposedReferencingColumnName] = referencingFieldDefinition.WithDirectives(directiveNodes);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Helper method to generate object definitions for linking entities. These object definitions are used later
+        /// to generate the object definitions for directional linking entities for (source, target) and (target, source).
+        /// </summary>
+        /// <returns>Object definitions for linking entities.</returns>
+        private Dictionary<string, ObjectTypeDefinitionNode> GenerateObjectDefinitionsForLinkingEntities()
+        {
+            IEnumerable<ISqlMetadataProvider> sqlMetadataProviders = _metadataProviderFactory.ListMetadataProviders();
+            Dictionary<string, ObjectTypeDefinitionNode> linkingObjectTypes = new();
+            foreach (ISqlMetadataProvider sqlMetadataProvider in sqlMetadataProviders)
+            {
+                foreach ((string linkingEntityName, Entity linkingEntity) in sqlMetadataProvider.GetLinkingEntities())
+                {
+                    if (sqlMetadataProvider.GetEntityNamesAndDbObjects().TryGetValue(linkingEntityName, out DatabaseObject? linkingDbObject))
+                    {
+                        ObjectTypeDefinitionNode node = SchemaConverter.GenerateObjectTypeDefinitionForDatabaseObject(
+                                entityName: linkingEntityName,
+                                databaseObject: linkingDbObject,
+                                configEntity: linkingEntity,
+                                entities: new(new Dictionary<string, Entity>()),
```

Comment:
No, `entities` is only referenced when we process relationships for an entity. But we don't process relationships for linking entities. Similarly `rolesAllowedForEntity`, `rolesAllowedForFields` are instantiated as empty dictionaries here because we anyways ignore all AuthZ conditions for linking entity and go on to add every field's definition to the linking object definition.